components/autocomplete/index.js

```diff
@@ -247,88 +299,67 @@ export class Autocomplete extends Component {
 	}
 
 	search( event ) {
-		const { open: wasOpen } = this.state;
+		const { open: wasOpen, suppress: wasSuppress } = this.state;
 		const { completers } = this.props;
 		// ensure that the cursor location is unambiguous
 		const container = event.target;
 		const cursor = this.getCursor( container );
 		// look for the trigger prefix and search query just before the cursor location
 		const match = this.findMatch( container, cursor, completers, wasOpen );
 		const { open, query, range } = match || {};
-		// create a regular expression to filter the options
-		const search = open ? new RegExp( escapeRegExp( query ), 'i' ) : /./;
 		// asynchronously load the options for the open completer
 		if ( open && ( ! wasOpen || open.idx !== wasOpen.idx ) ) {
 			this.loadOptions( open.idx );
 		}
+		// create a regular expression to filter the options
+		const search = open ? new RegExp( '(?:\\b|\\s|^)' + escapeRegExp( query ), 'i' ) : /./;
+		// filter the options we already have
+		const filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
+		// check if we should still suppress the popover
+		const suppress = open && wasSuppress === open.idx ? wasSuppress : undefined;
```

Comment:
I suggest adding parentheses to make eval order really obvious, instead of relying on precedence rules. One level is enough: `( a && b === c ) ? d : e`

Reply:
Done